tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml

```diff
@@ -0,0 +1,83 @@
+summary: Refresh and revert the gadget and kernel snap
+
+description: |
+    This test validates the kernel and gadget snaps can be refreshed
+    and reverted to the new snaps published to edge channel.
+
+systems: [ubuntu-20.04-*]
+
+environment:
+    CORE_CHANNEL: beta
+    CORE_REFRESH_CHANNEL: edge
+    BUILD_SNAPD_FROM_CURRENT: false
+    USE_CLOUD_INIT: true
+    ENABLE_SECURE_BOOT: true
+    ENABLE_TPM: true
+    SNAP/kernel: pc-kernel
+    SNAP/gadget: pc
```

Comment:
I think we also want these tests for snapd and core20 too, no?

Reply:
It was requested just to make it for those snaps, but it should be really easy to extend it to snapd and core20. @pedronis is it ok for you?

Reply:
yes, though not blockers, of the two snapd is higher priority, while core20 for sure could be done as a follow up

Reply:
@pedronis nice, thanks